### PR TITLE
Add --print-config flag

### LINKED ISSUE #169: Allow GitOps integration
Currently, the script generates and installs ASM config, which is not suitable for GitOps integration.  Ideally, users should treat their config as code, and review a diff of all changes before deploying to production via GitOps.  I think all that is needed here is a way to dump the config to stdout instead of applying it directly.

### PR DESCRIPTION
**Pull request by zerobfd**
Instead of installing anything, just print the config that would have been applied. This also skips verification and cluster/project setup. Closes #169.

Sample output (citadel was chosen to include an additional yaml file to overlay):

```
$>install_asm -l us-central1-c -n basic-suite-981hztoe -p test-cluster -m install -c citadel --print-config > testfile
$>cat testfile
---
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  profile: asm-gcp
  hub: gcr.io/asm-staging-images/asm
  tag: 1.7.1-asm.0
  components:
    pilot:
      k8s:
        env:
        - name: SPIFFE_BUNDLE_ENDPOINTS
          value: "test-cluster.svc.id.goog|https://storage.googleapis.com/mesh-ca-resources/spiffe_bundle.json"
        - name: ENABLE_STACKDRIVER_MONITORING
          value: "true"
  meshConfig:
    defaultConfig:
      proxyMetadata:
        GCE_METADATA_HOST: "metadata.google.internal"
        TRUST_DOMAIN: "test-cluster.svc.id.goog"
  values:
    global:
      trustDomain: "test-cluster.svc.id.goog"
      sds:
        token:
          aud: "test-cluster.svc.id.goog"
---
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    pilot:
      k8s:
        env:
        - name: SPIFFE_BUNDLE_ENDPOINTS
          value: ""
        - name: TOKEN_AUDIENCES
          value: "istio-ca,test-cluster.svc.id.goog"
        - name: ENABLE_STACKDRIVER_MONITORING
          value: "true"
  meshConfig:
    defaultConfig:
      proxyMetadata:
        TRUST_DOMAIN: "cluster.local"
        CA_PROVIDER: "Citadel"
        PLUGINS: ""
  values:
    global:
      trustDomain: "cluster.local"
      caAddress: ""
      pilotCertProvider: "istiod"
```

### CODE REVIEW

**Review by backward-compatible — APPROVED** [review]
LGTM

**Review by elfinhe — APPROVED** [review]

File: scripts/asm-installer/install_asm [hunk 1]
```diff
@@ -214,6 +225,11 @@ FLAGS:
   -e|--enable_apis                    Allow this script to enable necessary APIs
                                       on your behalf. Without this flag, it will
                                       abort if APIs are not already enabled.
+     --print_config                   Instead of installing ASM, print all of
+                                      the compiled YAML to stdout. All other
+                                      output will be written to stderr, even if
+                                      it would normally go to stdout. Skip all
```
Comment (djazayeri) [inline scripts/asm-installer/install_asm:231]: Not sure I agree about "skip all validations" in this case. Is that standard/idiomatic for this kind of command? We can discuss offline.

Comment (zerobfd) [reply scripts/asm-installer/install_asm:231]: I agree that it's a trade-off. In the context of the linked issue (GitOps / CI/CD integration) I thought it was safer for now to disable it by default. I don't know if the CI/CD user/service account is going to have the access to check everything, if the cluster names may be templated in later, etc. @therealmitchconnors do you have any guidance one way or the other?

Comment (therealmitchconnors) [reply scripts/asm-installer/install_asm:231]: Is this validation against user-supplied config, or against runtime values in the cluster or project?

Comment (zerobfd) [reply scripts/asm-installer/install_asm:231]: Runtime values. The CLI options/dependencies are still verified.

Comment (therealmitchconnors) [reply scripts/asm-installer/install_asm:231]: Then it does seem like a runtime validation might not be possible in some CI scenarios.  Perhaps we should have a separate flag to control this?

Comment (zerobfd) [reply scripts/asm-installer/install_asm:231]: Sure. Added #176 to track the extra flag.

**Pull request merged**